File: 128-longest-consecutive-sequence/longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:


        output = set(nums)

        longest=0

        for i in output:

            if (i-1) not in output:

                length=0

                while length+i in output:
                    length+=1

                longest= max(longest,length)

        return longest

                

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.ts
function longestConsecutive(nums: number[]): number {

    const set = new Set<number>();

    for(let i of nums){

        set.add(i)
    }
    let longest =0;

    for(let i of set){

        if(!set.has(i-1)){

            let length=1

            while(set.has(i+length)){

                length++;
            }

            longest = Math.max(longest,length)
        }
    }
    
    return longest
};